src/zepben/auth/authenticator.py

```diff
@@ -104,11 +109,11 @@ def fetch_token(self) -> str:
 
         return f"{self._token_type} {self._access_token}"
 
-    def _fetch_token_auth0(self, use_refresh: bool = False):
+    def _fetch_token_auth0(self):
         response = requests.post(
             construct_url(self.issuer_protocol, self.issuer_domain, self.token_path),
             headers={"content-type": "application/x-www-form-urlencoded"},
-            data=self.refresh_request_data if use_refresh else self.token_request_data,
+            data=self.refresh_request_data if self.use_refresh else self.token_request_data,
```

Comment:
maybe this needs to be `if self.use_refresh and self._refresh_token is not None` ?

Reply:
oh wait nope that's captured in fetch_token

Reply:
nvm it does need the and otherwise when fetching the access token it'll use the wrong refresh request data.